packages/zoe/src/contracts/attestation/expiring/expiringNFT.js

```diff
@@ -0,0 +1,143 @@
+// @ts-check
+
+import { makeStore } from '@agoric/store';
+import { AmountMath, AssetKind } from '@agoric/ertp';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { makeHandle } from '@agoric/zoe/src/makeHandle';
+import { validateInputs, mintZCFMintPayment } from '../helpers';
+import {
+  makeAttestationElem,
+  addToLiened,
+  hasExpired,
+} from './expiringHelpers';
+import { updateLien } from './updateLien';
+import { unlienIfExpired } from './unlienIfExpired';
+import { extendExpiration as extendExpirationInternal } from './extendExpiration';
+
+// Adds an expiring lien to an amount in response to an incoming call.
+// Can be queried for the current liened amount for an address.
+
+const { quote: q } = assert;
+
+/**
+ * @param {string} attestationTokenName - the name for the attestation
+ * token
+ * @param {Amount} empty - an empty amount in the external brand (i.e.
+ * BLD) that the attestation is about
+ * @param {ContractFacet} zcf
+ * @returns {Promise<{disallowExtensions: DisallowExtensions, addExpiringLien: AddExpiringLien, getLienAmount:
+ * GetLienAmount, makeExtendAttInvitation: MakeExtendAttInvitationInternal, issuer: Issuer, brand: Brand}>}
+ */
+const setupAttestation = async (attestationTokenName, empty, zcf) => {
+  assert(AmountMath.isEmpty(empty), `empty ${q(empty)} was not empty`);
+  const zcfMint = await zcf.makeZCFMint(attestationTokenName, AssetKind.SET);
+  const {
+    brand: attestationBrand,
+    issuer: attestationIssuer,
+  } = zcfMint.getIssuerRecord();
+
+  const externalBrand = empty.brand;
+
+  // Note: `amountLiened` is of the brand `externalBrand`
+
+  /** @type {Store<Address,Array<ExpiringAttElem>>} */
+  const lienedAmounts = makeStore('address');
+
+  const cannotGetExtension = new Set();
+
+  // This request *must* come from the owner of the address. Merely
```

Comment:
This seems to imply that some (many?) of these methods must only be called in the context of an address.

Instead of exposing functions with that constraint one-by-one, would you be able instead to provide a `makeAddressFacet(address) { return ...bundle of methods }` in this function's return result.  This prevents having to rely on the caller to manually do the currying and have to create its own facet from its (probably incomplete) understanding of the "address-specific" constraint.

Reply:
> Instead of exposing functions with that constraint one-by-one,

I see that this already works at the top level of the `creatorFacet`, but I think the pattern should be pushed down to reduce inter-file API dependencies.

Reply:
So I think there may be a misunderstanding about what this implies. There is only one function in `expiringNFT` and one in `returnableNFT` (`addExpiringLien` and `addReturnableLien`) that should only be called by the owner of the address. The other functions are called by cosmos or the general public. 

So if there's only one function with that requirement, I don't think we're consolidating much by making an object with one method which already has the address embedded. 

But, is there something I'm missing? How would you see this working?

Reply:
Actually, let me try it even with just the one method per object and see. The intended usage might be much clearer once it's rearranged into ocap form.

Reply:
Ok, just tried it, and I don't think it's a clear win. I think I will take @Chris-Hibbert's advice about the comments regarding the attMaker to ensure that it only gets handed to owner of the underlying assets. @michaelfig, if this change sounds like something you're still interested in (and maybe I'm misunderstanding what you mean), please feel free to change this contract, even after it's merged.

Reply:
Okay, thanks for exploring this.  Resolving.